src/LykkePublicAPI/Controllers/CompanyController.cs

```diff
@@ -32,15 +36,20 @@ IRegistrationsInfoCacheService registrationsInfoCacheService
         public async Task<CompanyInfoModels> GetOwnershipStructure()
         {
             var tradingWalletCoins = await _marketCapitalizationService.GetCapitalization(LykkeConstants.LykkeAssetId) ?? 0;
+
+            double treasuryAmount = 0;
+            foreach (var address in _companyInfo.LkkTreasuryWallets)
+                treasuryAmount += await _ninjaService.GetBalance(address, LykkeConstants.LykkeAssetId);
```

Comment:
It could be written as:

```c#
var treasuryAmountTasks = _companyInfo.LkkTreasuryWallets.Select(address => _ninjaService.GetBalance(address, LykkeConstants.LykkeAssetId));
var treasuryAmount = (await Task.WhenAll(treasuryAmountTasks)).Sum(t => t.Result);
```

Thus balances of all wallets will be abtained in parallel.

Reply:
Revised this way. But is still displayed here for some reason.